Reader text

rrrutledge opened the pull request:
This PR encompasses the proposed reader text of this Oriole.

If you want to actually see how this content looks in an Oriole, then [follow these instructions](http://oreillymedia.github.io/production-resources/oriole/#tips) for running the Oriole locally.

View sample Orioles [here](https://www.safaribooksonline.com/oriole/) to see what the format is like on other, finished products.

rrrutledge committed:
Draft text

rrrutledge commented on main.html:
```diff
@@ -1,77 +1,252 @@
-<h1>A Simple Oriole</h1>
```
All of this red text is the starter "template" text.  You can just ignore it.  Only look at the green (added) text.

rrrutledge committed:
Finished incorporating all notes into the text

Remaining notes are for non-textual items (e.g. speaker stories, diagrams)

rrrutledge committed:
Merge pull request #4 from rrrutledge/draft/text-nopre

Finished incorporating all notes into the text

rrrutledge commented on main.html:
```diff
@@ -1,77 +1,240 @@
+    The first word <em>Voluntary</em> means that engagement in InnerSource occurs of their own free will from both the guest and host teams.
+    The guest team voluntarily donates code to the host team and the host team voluntarily accepts it.
+    The opt-in nature of this setup means that each side needs to be certain that their involvement adds value to the others' objectives.
+    Never is a host team required to accept a contribution that isn't in ultimate alignment wit their overall mission.
```
typo: with

rrrutledge committed:
Update main.html

Fix typo: wit -> with

rrrutledge committed:
Merge pull request #5 from rrrutledge/rrrutledge-patch-1

Update main.html

rrrutledge commented:
View sample Orioles [here](https://www.safaribooksonline.com/oriole/) to see what the format is like.

rrrutledge commented:
One big point - What is the correct form of the term (I am pretty sure I like InnerSource)?
* InnerSource
* inner-source
* innersource
* inner source
?

Tell me if there is contention about this point, but I think that whatever term there is should be usable as both a noun, verb, and an adjective (which is the ways I think you can use the term "open source").

rrrutledge commented on main.html:
```diff
@@ -1,77 +1,240 @@
+    Imagine separate teams producing separate pieces of software with one team's software depending on that of another.
+    This situation is common in a large enterprise where a single team producing software may have dozens or hundreds of consumers.
+    When consuming teams need changes, normally the producing team has some sort of requirements prioritization process to decide which they will work on.
+    For feature requests that are not priorized, the consuming team is faced with 3 choices, each of which come with their own drawbacks.
```
It may not pop out from an initial reading of the document, but each of these options intentionally has 3 syllables with the same syllabic emphasis pattern.  Just my opinion, but I think that this symmetry makes them easy to speak and natural to substitute for one another in sentences just as the actions that they represent are, in some way, substitutes for each other in real life.

rrrutledge committed:
Roll your own -> Work around

Avoiding cigarette-connotated phrase

rrrutledge committed:
Merge pull request #6 from rrrutledge/draft/text-rollyourown

Roll your own -> Work around

rrrutledge commented:
Some terms that I am trying to use intentionally and consistently.  Every time the concept is mentioned it should be done with these chosen words.
* InnerSource
* Producer / Producing team
* Consumer / Consuming team
* Feature request / Requested feature

rrrutledge committed:
Cleaning up wording

rrrutledge committed:
Additional edits

rrrutledge committed:
Spelling changes

Danese reviewed (commented):
In the text above this introduction of HOST and GUEST, you talk about the "consuming team" quite a lot. Might be good to map that to this section?

rrrutledge committed:
Removed notes for speakers' script

Moving into [dedicated PR](https://github.com/rrrutledge/innersource-intro/pull/7).

rrrutledge commented on main.html:
```diff
@@ -1,77 +1,255 @@
-plt.plot(x, np.sin(x), x, np.cos(x));
+<p>
+  InnerSource is the methodology of applying open source practices and principles to internal software development.
+  This methodology provides a low-friction process for ensuring that the features and functionality of shared projects are responsive and nimble to the changing needs of their many internal consumers.
```
A little long and business-speaky

rrrutledge replied:
Made a change.

dowdell commented on main.html:
```diff
@@ -1,77 +1,255 @@
+  With those definitions, here is the basic outline of an InnerSource contribution.
+</p>
+<p>
+  <ol>
```
I don't think an `<ol>` is allowed to be a child of a `<p>`, ref: https://stackoverflow.com/a/10601411

rrrutledge replied:
Fixed, Nick.  Thanks for keeping the review technical ❤️ 👍 .

dowdell commented on main.html:
```diff
@@ -1,77 +1,255 @@
+<p>
+  In addition to this alignment, there is also a grass-roots evangelism that takes hold in many contributors.
+  Having made a code contribution, they feel a stake in the host team project.
+  It's something that they recommend to their colleagues or new teams that they join.
```
+1

dowdell commented on main.html:
```diff
@@ -1,77 +1,255 @@
+  The principles are:
+</p>
+
+<ul>
```
This one looks good!

dowdell reviewed (approved):
LGTM, with one small HTML-related nitpick.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+  Finally, we'll share other references that can help you deepen your understanding and involvement in the craft of InnerSource.
+</p>
+
+<h2>Context</h2>
```
Context for what? I thought the introduction was the context?

mrsanz replied:
Perhaps frame this as 
>A common inner-source scenario?

rrrutledge replied:
Done.

mrsanz replied:
looks good!

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+  Imagine two teams delivering separate pieces of software with one team's software depending on that of the other.
+  This situation is common in a large enterprise where a single team producing software may have dozens or hundreds of consumers.
+  When consuming teams need many features, producing teams normally have some sort of requirements and prioritization process to decide which features they will work on.
+  For feature requests that are not prioritized, the consuming team is faced with 3 choices, each of which come with their own drawbacks.
```
Why not include the 4th option here (collaboration).  Sometimes it does come up organically.

rrrutledge replied:
Well, the 4th option (inner source) is the one that we're introducing later on.  If people are already collaborating in this way then they don't necessarily have need of this training.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+<ol>
+  <li>
+    <p>
+      <strong>Wait it out</strong>.
```
Might call it out at the end of this section `Wait it out` that often a solution is not found or that folks implement a workout.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+  <li>
+    <p>
+      <strong>Workaround</strong>.
+      A consuming team that doesn't want to wait may do extra work somewhere else to compensate for the absence of their requested feature.
```
Do a find and bold + italic all of the terms.  in this case I think you defined `consuming team`

mrsanz replied:
Do this for other key terms that you define as well.

rrrutledge replied:
I haven't actually come out and defined it in a straightforward manner.  Do you think I should?

mrsanz replied:
In looking at the html it looks fine without this change.  I would agree since you didn't come out and define it, it doesn't need to be highlighted in this way.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+    <p>
+      <strong>Workaround</strong>.
+      A consuming team that doesn't want to wait may do extra work somewhere else to compensate for the absence of their requested feature.
+      This extra work may come as a modification (hack) in the consuming project.
```
flip this?  hack (modification) use the jargon first then define the jargon.

mrsanz replied:
or avoid jargon.

rrrutledge replied:
Avoided.

mrsanz replied:
I like it!

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+      Alternatively, the consuming team may create a new project that meets their needs and replaces their use of all or part of the producing team's system (code/project duplication).
+    </p>
+    <p>
+      <em>Workaround</em> is suboptimal from a variety of standpoints.
```
Perhaps instead of going into why this is sub optimal.  Show how this only extends so far.  Eventually critical mass wins, and the workaround is pushed to the side or deprecated by the **_host team_** for new and shiny features.

rrrutledge replied:
I see that point not as a replacement for the other points here, but more like extra information to the 3rd bullet point about why duplicate projects don't make sense.  What do you think?

mrsanz replied:
I like the change!  LGTM!

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+      </ul>
+    </p>
+    <p>
+      Despite these drawbacks, one big win for the consuming team is their ability to get the requested feature via their own efforts only (no hard dependencies).
```
Highlight this first.

mrsanz replied:
not sure how the `(no hard dependencies)` relates precisely.  I can guess as to what this means because I have more context but a novice surely won't understand how  dependencies affect the larger system.   

This might be a nice topic to talk about toward the beginning in talking about how this topic can be thought of in terms of dependencies and what the contracts look like between teams for dependencies..

rrrutledge replied:
> Highlight this first

I think it would be fine to do that, but just curious why put it first?

mrsanz replied:
I think you should put it first because if this is the most common result of the **Workaround** scenario then it should be highlighted first rather than at the end.  

In thinking about this more I think all 3 scenarios could have a little more structure to them to more clearly present the material.  

Suggested structure: Features, Advantages (Disadvantages), Benefits.
[This article](http://devedge-internet-marketing.com/2012/10/06/features-advantages-and-benefits-fab-statements/) goes into more detail.

Doing this will help the readers be able to compare and contrast the three scenarios.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+  </li>
+  <li>
+    <p>
+      <strong>Escalate</strong>.  The consuming team may not take "no" for an answer and, instead, advocate to someone in the producers' management hierarchy to influence (or force) the producing team to do the work.
```
Where is this pattern more common typically found?  In big companies with strong management chains and strong silos?

rrrutledge replied:
I don't know, but are you intimating that information needs to be here?

mrsanz replied:
Since the audience for this is broad, I think it may make sense to call out  Where to be on the look out for these things patterns.

I would recommend adding something like:  The **Escalation** pattern can most often be found where there is a lot of direct control in the end result, or where teams operate in silos.  

You may consider doing this for the scenarios as well.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+      <strong>Workaround</strong>.
+      A consuming team that doesn't want to wait may do extra work somewhere else to compensate for the absence of their requested feature.
+      This extra work may come as a modification (hack) in the consuming project.
+      Alternatively, the consuming team may create a new project that meets their needs and replaces their use of all or part of the producing team's system (code/project duplication).
```
clarification re: `(code/project duplication)`  do you mean that this Workaround pattern creates a lot of code and project duplication.  I think the point deserves more than an aside.

rrrutledge replied:
It's bullet #3 (below).  Is that enough call-out?

mrsanz replied:
Yea I think I was struggling with the structure again.  See the note I added about a `Features, Advantages, Benefits` format for these.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+    </p>
+    <p>
+      This option sounds attractive for the consuming team because they get the requested feature without doing the work to implement or maintain it.
+      It is still a drag on the team, though, because it necessarily diverts some of their attention and work to the non-productive task of escalation.
```
Also the work is often done to the wrong specification. Meaning that while the "official" capability is there, an approved right contract may not be created.  Thus the consuming team will still have to use the **_Workaround_** method to get this to work after time has been wasted by the original team.  

This ends up creating friction between teams because the **_host team_** will point to the work and say "its done", while the **_consuming team_** will point to the work and say "I can't use that!"

rrrutledge replied:
While that's true, it's a situation that is present also even in cases where the **host team** accepts the work from the beginning without any form of escalation.  So I don't think that it belongs in this section.

mrsanz replied:
ok!

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+</ol>
+
+<pre>
+  The "Escalate" option could be a place to include work for Danese's "cheeses" diagram.
```
Perhaps instead come up with a series of diagrams describing each of these scenarios coming up with a similar anecdote modeled after the cheese diagram.  

I'm sure there are other similar analogies that can resonate with people.

rrrutledge replied:
Good idea!  Captured that in the list of proposed graphics.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+<h2>Solution Overview</h2>
+
+<p>
+  This Oriole explains a fourth option available to the consuming team - <strong>InnerSource</strong>.
```
The beginning of this section 

>This Oriole explains ...

seems oddly placed here.  Do you expect people to jump right into this part of the content without reading the rest?

rrrutledge replied:
No.  I can change.

rrrutledge replied:
Removed the sentence.

mrsanz replied:
super!

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+
+<p>
+  This Oriole explains a fourth option available to the consuming team - <strong>InnerSource</strong>.
+  This strategy provides the wins of <em>Workaround</em> and <em>Escalate</em> without the same drawbacks to the teams and company.
```
So far you've primarily attached negative connotations to Workaround and Escalate.  

Might be better to say something like:
> This strategy provides the benefits x, y z without the drawbacks associated with the **_Workaround_** and **_Escalation_** patterns to the team and company.

rrrutledge replied:
There are some positive results of `Workround` and `Escalate` that I've called out.  Maybe I just need to make those more clear in their sections?  What do you think?

mrsanz replied:
Your introduction reads
> InnerSource is the methodology of applying open source practices and principles to internal software development.

perhaps start with that.  Suggestion -> `Innersource as a methodology provides the wins of Workaround and Escalation without the drawbacks.`

I don't think you need to mention who the players are (you just spent a good amount of time outlining scenarios), and I think that you should stick to the same language as in the introduction for this part.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+  This strategy provides the wins of <em>Workaround</em> and <em>Escalate</em> without the same drawbacks to the teams and company.
+</p>
+<p>
+  The basic idea of <em>InnerSource</em> is this: If a consuming team can't get what it needs by submitting a feature request then it may submit a pull request instead.
```
While this is true from the perspective of **_consuming teams_**.  But it also needs to be true that the **_host team_** needs to have the attitude that they're putting in engineering time and effort in a direction we want to go, so lets help them get there.

I think that it might be better to frame the attitude or principles of innersource source then dig into the mechanics.

rrrutledge replied:
> this is true from the perspective of consuming teams

Most of this document is written from that perspective.  I think there is another document to write from the perspective of the ***host team***, but it would be too much to include that all in one article.  Theoretically there will be an Oriole for training product owners coming later on, and that training could be a good place to include more detail on the perspective and benefits to the ***host team***

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+</p>
+<p>
+  The basic idea of <em>InnerSource</em> is this: If a consuming team can't get what it needs by submitting a feature request then it may submit a pull request instead.
+  The producing team can partner to review and accept the pull request and the functionality it implements.
```
Really they _must_ review and accept it for innersource to work well.

rrrutledge replied:
Made the language more affirmative.

mrsanz replied:
Affirmative!

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+  First, in the context of the code contribution process, we call the team that is receiving the code the <strong>Host team</strong>.
+  Conversely, the team giving the code is the <strong>Guest team</strong>.
+  There are also a few key individuals to name.
+  First, the <strong>Product Owner</strong> determines what functionality the host team is willing to accept as contributions.
```
You could generalize this a bit. (a lot of Firsts)

> We'll also call the group that determines what functionality the **_host team_** is willing to accept as contributions, the **_Product Owner_**.

rrrutledge replied:
Down to just one `First`

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+</p>
+
+<p>
+  With those definitions, here is the basic outline of an InnerSource contribution.
```
A diagram seems to be better suited for this content.

mrsanz replied:
I'm pretty sure you have one you could adapt.

mrsanz replied:
Also I think in practice you focus on the output as a **_contract_**.  It might be nice to touch on that.

rrrutledge replied:
Added this suggestion to the list of proposed diagrams.  *contract* is definitely mentioned below - is that sufficient?

mrsanz replied:
I was thinking of the term contract the way we use it.  Not always as a data contract but even as granular as the contract of the AC.  

> describe the requested feature in terms agreeable to the guest team


is more along the lines of what I mean by contract in this context.

I still think there could be more emphasis (possibly an oral anecdote)  This contract is often where teams get it wrong. 

Perhaps a sentence like:  `Consider all of the work that you do in setting up the work for to collaborate on as though it were a signed agreement and commitment from each team.` after the outline.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+<p>
+  Note that these steps do not assume a specific system for the general organization of a team's time or priorities.
+  InnerSource assumes that teams already have existing such methods of organization.
+  InnerSource simply provides a framework of how to use them to work together where there is a guest team desiring to contribute code to a host.
```
_**Simple**_ is using the workaround solution!  😁   If this is simple then I'd like to see it described in simpler terms.

Sentences like 
> Trusted committers' time is made available to the guest team when the contributor is working.

Are not simple to parse when I've just been introduced to the terms **_Trusted Committer_** and **_Guest Team_**.

rrrutledge replied:
Take another look after I've incorporated many of your suggestions?

mrsanz replied:
Yea I like it.  Thanks!

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+
+<p>
+  There are many benefits to collaborating via InnerSource.
+  First and foremost, <strong>the guest team gets their feature</strong> request when they need it without the long-term burden of maintaining it.
```
**_guest team_**?  It was **_consuming team_** earlier.  

> The team requesting the feature gets their request fulfilled in the most ideal timeframe, without the long-term burden of maintenance.

Oh and they get access to experts!

rrrutledge replied:
I think that after the term of **guest** is introduced it is used exclusively.

rrrutledge replied:
Updated some text later on in this section calling out the benefit of contributors getting to work with trusted committers.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+<p>
+  There are many benefits to collaborating via InnerSource.
+  First and foremost, <strong>the guest team gets their feature</strong> request when they need it without the long-term burden of maintaining it.
+  When the guest team would have spent time on the feature regardless (whether through <em>Workaround</em> or <em>Escalate</em>), InnerSource provides a way to use that time towards developing the feature in a place that's centrally-maintained and shared.
```
Call out the benefit clearly first?  Time saved!  The code is centrally shared!

rrrutledge replied:
Rewritten.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+</p>
+
+<p>
+  While that result is the shining benefit of InnerSource, another result that's easy to overlook is the fact that <strong>the host team gets their feature, too</strong>.
```
Why is this easy to overlook?  This whole Oriole should be about a **_guest team_** getting a feature completed with **_a host team_**.  If it isn't clear by this point I think that the message should be more focused on this point.  

Getting internal features complete with as little friction as possible where there are cross team dependencies should be the main goal of Innersource!

rrrutledge replied:
I agree, just my own experience in-practice feels like **host** teams show little interest in getting features in their backlog completed by guests.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+  Given the fixed capacity of the full-time members of the host team, it's likely that at times the combined business roadmaps of their consumers will require very (or even unreasonably) large amounts of work to be done in the host team's product(s).
+  Without InnerSource, this situation easily leads to a stressed, overworked team dealing with many feature requests escalated to its leaders.
+  However, if the host team operates via InnerSource, the engineering resources required to build those features will appear in-proportion to their importance in the form of guest contributors.
+  InnerSource becomes a force multiplier that allows a host team to temporarily act larger than its actual size during times of high demand.
```
I like this point!  Is there a way to add emphasis in the reader to this with an example or something?

rrrutledge replied:
Not sure?

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+</p>
+
+<p>
+  Beyond the raw work that the host team is able to accomplish in its system, regular InnerSource contributions give the host team better requirements and prioritization alignment with all of its consumers.
```
I'd like you to highlight this point more when describing the process in earlier sections.  If its an afterthought it won't happen., and it seems critical that part of the process is about aligning on requirements or contracts between teams.

rrrutledge replied:
🤔 Not sure exactly what to do here.  I feel like this result just happens as a result of regularly following the inner source process.

mrsanz replied:
I think I'd like to see a 1,2,3 punch here...

Main benefits
1. The guest team gets their feature
2. The host team gets their feature, too!
3. The host team gets better requirements and prioritization alignment with all of its consumers

Additional Benefits
1. Cheaper maintenance burden, more than one team can troubleshoot and the support the software.
2. Distributed knowledge, as there is not a single point of failure and less tribal knowledge floating around the organization.
3. Clear responsibility, as there is a stated contract of how teams are working together.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+</p>
+
+<p>
+  In addition to this alignment, there is also a grass-roots evangelism that takes hold in many contributors.
```
I think that this is often overlooked.  ;)   The most effective path to Innersouce is via grass-roots evangelism.

rrrutledge replied:
Great!

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+  In addition to this alignment, there is also a grass-roots evangelism that takes hold in many contributors.
+  Having made a code contribution, they feel a stake in the host team project.
+  It's something that they recommend to their colleagues or new teams that they join.
+  They understand the project better and are able to answer questions about it to others, relieving the host team of some of that burden.
```
This also helps eliminate the tribal nature of siold company cultures.  Might be worth a mention.

rrrutledge replied:
Sure.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+</p>
+
+<p>
+  While this training contains a high-level overview of InnerSource, there are many more details useful in making InnerSource actually work for your team.
```
This sentence made me think that it might be nice to open with 
> You should learn these things in this Oriole:
> What problem does innersourcing solve
> The two main principles of innersource
> Am I doing Innersource already?
> What a basic innersource pattern looks like. 
> Innersource anti-patterns

Something along those lines.  It'd be nice to know what I'm getting myself into at the outset and what I should expect to walk away with.

rrrutledge replied:
There was an attempt to do that with this paragraph:

```
  In this Oriole, we'll define some key terms used when discussing InnerSource.
  We'll walk through the high-level mechanics of how InnerSourcing can work and enumerate the key principles upon-which these mechanics are based.
  Finally, we'll share other references that can help you deepen your understanding and involvement in the craft of InnerSource.
```
Maybe share what needs to change about that introduction to communicate the points that you'd like to see?

mrsanz replied:
ok! I'll comment on the main body of text...

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+
+<p>
+  While this training contains a high-level overview of InnerSource, there are many more details useful in making InnerSource actually work for your team.
+  We're working hard on at least 3 more Orioles to provide some of this information, so stay tuned for those.
```
Unfortunately, I didn't think there was enough meat to want to keep on with another Oriole.  Orioles (like reading a book) can be a decent time investment to really learn the material.  I had to work pretty hard to understand the concepts and I didn't see how they were immediately applicable to real life.  I think if the concepts were a bit easier to immediately put into action, then I'd consider looking at the other Orioles.

mrsanz replied:
Consider removing 'working hard' for this advertisement.  I'm not sure that the call to action needs the extra emphasis.

rrrutledge replied:
> removing 'working hard' 

Removed the whole sentence.

rrrutledge replied:
> I didn't think there was enough meat to want to keep on with another Oriole

I took a lot of your suggestions.  Have another look and see what you think?  If this conclusion is still the same then share what kinds of details could make these concepts more applicable to real life?

mrsanz replied:
It is better without the sentence.  I enjoyed it much more this time!

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+
+<p>
+  It's critical that this mentorship for contributors is <em>Prioritized</em> by the host team.
+  Trusted committer availability to mentor is made available to guest team contributors <em>at the time that they need it</em> as opposed to when it is convenient for the trusted committer and host team.
```
How does that work with async workflows?  `At the time`, implies immediate but in reality there is at least a 4-8 hour lag due to async workflows between team members and even team cadences.

rrrutledge replied:
I suppose it's not always guaranteed to be immediate, but the _strive_ to make it so.  Updated wording.

mrsanz replied:
I like what we came up with today.  Even going as far as saying "shepherding in contributions"

mrsanz replied:
its better though.. no changes needed!

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+  </ol>
+</p>
+<p>
+  Note that these steps do not assume a specific system for the general organization of a team's time or priorities.
```
Seems like there should be something in here about conflict resolution as well.  I know that this can come up and is what makes this part.

rrrutledge replied:
I'm not sure what you mean.  Can you give more detail?

mrsanz replied:
I was trying to get at the fact that some basic conflict resolution needs to take place in order to remove some of the barriers to adoption.  Now that the solution detail I think this part is good as it is.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+    <li>
+      Supported by the trusted committer(s), the contributor submits the pull request(s) to implement the requested feature.
+    </li>
+    <li>
```
Can you distill these down to 3 steps from 9?

>  feature request (by guest team)
>  request review (by host team)
>  request prioritized (by both teams together, determining who does the work)

or something a bit more simplified than the 9 steps

I'd like to see the detail in future orioles or captured in the reader or in sub sections.  I think the initial workflow is a bit hard to understand when presented in a list format.

rrrutledge replied:
Good idea.

rrrutledge replied:
Down to 3.

mrsanz replied:
yay!!!! 🌮    Looks good!

mrsanz commented on main.html:
```diff
@@ -1,77 +1,255 @@
+</p>
+
+<h2>
+  Benefits
```
consider breaking `Benefits` into two sections, benefits to the **_guest team_** and benefits to the **_host team_**

rrrutledge replied:
Considered, and re-wrote a little bit, but not so distinct as you're suggesting.  Have another look?

mrsanz replied:
LGTM

rrrutledge commented on main.html:
```diff
@@ -1,77 +1,255 @@
+<h2>Solution Detail</h2>
+<p>
+  Let's take a closer look at how the mechanics of InnerSource can work.
+  To aid in this explanation, we'll define a few key terms.
```
Could use a diagram that shows a picture of these key terms working together.

rrrutledge replied:
Captured in the list of proposed graphics.

rrrutledge commented on main.html:
```diff
@@ -1,77 +1,255 @@
+<ol>
+  <li>
+    <p>
+      <strong>Wait it out</strong>.
```
Could turn into "Wait forever" ...

rrrutledge replied:
Updated.

rrrutledge commented:
Some feedback from a group review session:
* The word "value" is only used here once.  Is there any more clarity that you can give to the value of inner sourcing?
* Another benefit of inner sourcing is that it will lead to a project that is easier to consume.
* What is the estimated cost to the host team (in terms of trusted committer and product owner time) to participate in inner source in this manner?
* Need to have the example situation be more specific.  Still needs to be kept generic enough so that it is broadly-applicable, though.  One suggestion: A client team needs a field added to an API response.

rrrutledge committed:
PR feedback

rrrutledge committed:
guest/host

rrrutledge committed:
PR feedback

rrrutledge committed:
Merge pull request #8 from rrrutledge/draft/text-pr

Lots of PR feedback!

rrrutledge reviewed (commented):
Addressed a lot of comments and left responses on most.  Have another look!

rrrutledge committed:
Remove preformatted text/notes

rrrutledge commented:
> In the text above this introduction of HOST and GUEST, you talk about the "consuming team" quite a lot. Might be good to map that to this section

Added a sentence to try to make that associated more explicit.

rrrutledge commented:
Here is [a PDF](https://github.com/rrrutledge/innersource-intro/files/1890103/Introduction.to.Inner.Source.Oriole.pdf) of the current text so that you don't have to run the _Docker_ container in order to read it.

utanapishtim reviewed (approved):
Really good!

mrsanz commented on main.html:
```diff
@@ -1,77 +1,231 @@
+</p>
+
+<p>
+  In this Oriole, we'll define some key terms used when discussing InnerSource.
```
I think that these three sentences are a bit too generic.  They don't have a lot of 'teeth'  It is too self-referential.
Suggestions:

Change "define some key terms used when discussing InnerSource" to "teach you how to identify scenarios where InnerSource Principles can be applied"

Add something like: "Be able to describe what InnerSource looks like"

rrrutledge replied:
> Change "define some key terms

🤔 Defining key terms is an explicit goal of this content.  Do you agree with [these proposed goals](https://github.com/rrrutledge/innersource-intro/blob/master/outline.md) for this project?  I want to check if we have different goals in mind as we comment and work on this material together.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,231 @@
+<p>
+  In this Oriole, we'll define some key terms used when discussing InnerSource.
+  We'll walk through the high-level mechanics of how InnerSourcing can work and enumerate the key principles upon-which these mechanics are based.
+  Finally, we'll share other references that can help you deepen your understanding and involvement in the craft of InnerSource.
```
Consider adding some teeth to this.  


`Finally, we'll share additional resources, so that you can get involved in the community that is forming around InnerSource.`

rrrutledge replied:
Suggestions?  I think the weight of this description matches the weight of the actual additional resources.  Are you saying that the description needs to carry more weight or the resources themselves aren't massive enough to call out in this way?

mrsanz replied:
The introduction is too buzz-wordy.  These things can be more specific focused on the perspective of the person going through the oriole.  

> Walk through high-level mechanics

What does this mean?   While I agree there are mechanics I'd like something practical I can do after this oriole.  Can I identify the mechanics and patterns? Anti-patterns?  etc.. something that I from the perspective of the participant can say I can do now that I couldn't before.  

>  we'll share other references 

what references? Why do I care about references at this point? I just started looking at this document.  Isn't this going to teach me about this stuff?  Seems odd so early on to setup pointers to reference material.  How do I get involved?  What are ways to engage others in the innersource community?  This is the type of action that should be called out instead I think.

>  deepen your understanding and involvement in the craft of InnerSource

My understanding?  Ok what am I going to learn?  Also I'm involved already?  I'm already participating by watching this oriole, how can you deepen my involvement?

rrrutledge replied:
Use more conversational (and specific) language.

rrrutledge replied:
Don't really need to introduce the inner source commons.  Or be more specific about it.

mrsanz reviewed (changes requested):
Three main pieces of feedback.

1) Use the pattern of `Features Advantages(Disadvantages), Benefits` when describing the scenarios.  This will help frame each scenario and allow your reader to consider each one as compared to the others

2) The Benefits Section is pretty wordy.  I'd consider introducing the benefits, providing them in a concise way (perhaps a list)  Then elaborate on each item as needed.  As it is now I have to hunt for each benefit.

3) Update the "Here's what you'll learn paragraph"

rrrutledge reviewed (commented):
> Use the pattern of Features Advantages(Disadvantages), Benefits when describing the scenarios

Curious for your thoughts on the following:
* I've been starting off each one with a definition of the scenario, too.  Or maybe that's what you mean by `Feature`?
* I really want to highlight how these are negatives, so I think that listing the disadvantages last makes sense to do that.

> As it is now I have to hunt for each benefit.

A few of the benefits I've called out inline in **heavily emphasized text**.  Would it work to do that for the other benefits so the flow of prose can be kept but also a scan of the text with your eyes can help you to pick out what are the benefits?

mrsanz commented:
> I really want to highlight how these are negatives, so I think that listing the disadvantages last makes sense to do that.

So in fact you don't do this for all of these scenarios.  (The Workaround section doesn't do this)  Here is a suggestion of one that pretty much follows the format already.

- a section outlining the "What" or Feature (also setting up the scenario).

The first scenario I'm calling the **Wait it out** scenario.  In this option, the consuming team requests a new feature and that feature for whatever reason is not picked up by the producing team.  The consuming team does nothing and limps along without the requested functionality, forced to **Wait it out**.

- a section outlining "how" it plays out. Highlighting Advantages and Disadvantages:

Sometimes, waiting for a solution to become more mature can make sense.  This might be fine if there isn't an urgency associated with a feature request or if the benefit of the feature request is a small one.  In these cases,  **Wait it out** might be just fine.

- finally a section outlining "why" this occurs.  Driving point home.  In the scenario where there are  Disadvantages, then Drawbacks can be highlighted instead of benefits. 

Waiting may also come with real amounts of pain for the consuming team, especially if the requested functionally is never delivered.  Be aware though that this can cost the company or individuals in ways that may not be obvious, affecting developer happiness or extra overhead for costs.

rrrutledge commented:
> Use the pattern of Features Advantages(Disadvantages), Benefits

We like: _Definition_, _Advantages/Disadvantages_, _Drawbacks_ for each of the antipatterns.
We like: _Definition_, _Advantages_, _Benefits_ for the inner source section.

rrrutledge committed:
PR feedback

rrrutledge commented:
Addressed all feedback.  Here is 
[a PDF view](https://github.com/rrrutledge/innersource-intro/files/1924845/InnerSource-Intro.pdf) of the content if that's easier to review.

mrsanz commented on main.html:
```diff
@@ -1,77 +1,233 @@
+      Alternatively, the consuming team may create a new project that meets their needs and replaces their use of all or part of the producing team's system (code/project duplication).
+    </p>
+    <p>
+      This strategy allows the consuming team to get the requested feature via their own efforts only (no hard dependencies).
```
Really minor nit here.  I don't know what a _hard_ dependency is.  Is this a _**direct**_ dependency?  That might be a better way to describe this aside since it is so short.  

Why not just include these parenthetical into the main idea?

mrsanz replied:
also there why are there not any parentheticals for the _**wait it out**_ scenario.

rrrutledge replied:
Updated.

mrsanz reviewed (approved):
Overall looks good.  Left 2 minor nit comments, other than that it is great!!!!  Nice job, I can't wait to run through the finished Oriole!

rrrutledge committed:
Hard dependency -> direct dependency
